File: src/main/java/com/dashboard/models/JPA/Task.java
package com.dashboard.models.JPA;

import com.sun.istack.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.validation.annotation.Validated;

import javax.persistence.*;
import java.util.Set;

@Data
@Entity
@Validated
@Table(name="task")
@NoArgsConstructor
@RequiredArgsConstructor
public class Task{

    @Id
    @GeneratedValue
    private Integer id;

    @Column @NonNull
    private String taskname;

    @Column @NonNull
    private String description;

    @Column
    @NonNull
    @Enumerated(EnumType.STRING)
    private Priority priority;

    @Column
    private int percentage = 0;

    @Column @NotNull @Enumerated(EnumType.STRING)
    private Status status = Status.NOT_READY;

    @ManyToMany(mappedBy = "tasks")
    private Set<Employee> assignedTo;
}

File: src/main/kotlin/models/jpa/Task.kt
package dashboard.models.jpa

import com.sun.istack.NotNull
import models.jpa.Priority
import models.jpa.Status
import javax.persistence.*

@Entity
class Task(
    @Id @GeneratedValue val id: Int? = 0,
    val taskname: String? = null,
    val description: String? = null,
    @Enumerated(EnumType.STRING) val priority: Priority? = Priority.NORMAL,
    val percentage: Int? = 0,
    @NotNull @Enumerated(EnumType.STRING) val status : Status? = Status.NOT_READY,
    @ManyToMany(mappedBy = "tasks") val assignedTo: Set<Employee>? = null
)